openvpn/middlewares/server/auth/middleware.go

```diff
@@ -0,0 +1,153 @@
+package auth
+
+import (
+	"fmt"
+	log "github.com/cihub/seelog"
+	"github.com/mysterium/node/openvpn"
+	"github.com/mysterium/node/openvpn/middlewares"
+	"net"
+	"regexp"
+	"strconv"
+)
+
+type middleware struct {
+	authenticator Authenticator
+	connection    net.Conn
+	username      string
+	password      string
+	clientId      int
+	keyId         int
+	state         middlewares.State
```

Comment:
Are all fields required to be struct fields? I.e. `password` seems only needed in `ConsumeLine` method - limiting it's scope to that method might simplify this struct.

Reply:
ConsumeLine works by analysing line by line. Thus it needs to store some parsed data as a state to be used for later time by the same method.